src/test/java/com/fasterxml/jackson/dataformat/xml/ser/XmlPrettyPrinterTest.java

```diff
@@ -188,4 +189,75 @@ public void testMultiLevel172() throws Exception
                 +"</Company>\n",
                 xml);
     }
+
+    public void testNewLine_withCustomNewLine() throws Exception {
+        Company root = new Company();
+        root.employee.add(new Employee("abc"));
+
+        String xml = _xmlMapper.writer()
+            .with(new DefaultXmlPrettyPrinter().withCustomNewLine("\n\rLF\n\r"))
+            .with(ToXmlGenerator.Feature.WRITE_XML_DECLARATION)
+            .writeValueAsString(root);
+        // unify possible apostrophes to quotes
+        xml = a2q(xml);
+
+        // with indentation, should get newLines in prolog/epilog too
+        assertEquals("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                + "<Company>\n\rLF\n\r"
+                + "  <e>\n\rLF\n\r"
+                + "    <employee>\n\rLF\n\r"
+                + "      <id>abc</id>\n\rLF\n\r"
+                + "      <type>FULL_TIME</type>\n\rLF\n\r"
+                + "    </employee>\n\rLF\n\r"
+                + "  </e>\n\rLF\n\r"
+                + "</Company>\n",
+            xml);
+    }
+
+    public void testNewLine_systemDefault() throws Exception {
+        Company root = new Company();
+        root.employee.add(new Employee("abc"));
+
+        String xml = _xmlMapper.writer()
+            .with(new DefaultXmlPrettyPrinter())
+            .with(ToXmlGenerator.Feature.WRITE_XML_DECLARATION)
+            .writeValueAsString(root);
+        // unify possible apostrophes to quotes
+        xml = a2q(xml);
+
+        // with indentation, should get newLines in prolog/epilog too
+        assertEquals("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
```

Comment:
Need to replace `\n` with system linefeed so test passes on Windows systems too.

Reply:
Right, done! 🙏🏼

And plus I checked other similar test cases' assertions and some hard coded `\n` as system linefeed. 
@cowtowncoder if you agree, I will make a separate PR on "replacing all hard coded `\n` to the value read from ` System.getProperty("line.separator");`"?

Reply:
Yeah sounds good wrt hard-coded linefeeds